### DISCUSSION TITLE
v0.2.0 Beta 2.0

### DISCUSSION THREAD

**Discussion by awtkns** [reworkd/AgentGPT#249] [Announcements]
<h1 align="center">
🎉🤖&nbsp;&nbsp;&nbsp;Beta 2.0&nbsp;&nbsp;&nbsp;🤖🎉
</h1>

We released [AgentGPT](https://github.com/reworkd/AgentGPT) a little over a week ago and have been blown away by the community's response! In a little over a week we have surpassed **750k+ users**, **16k+ stars**, and a discord community **10k+ strong**. Due to the insane reception of [AgentGPT](https://github.com/reworkd/AgentGPT) we had to focus the majority of this week on scalability. We are currently averaging over **1.5M+** tasks executed daily!

<img width="1236" alt="Task Executions" src="https://user-images.githubusercontent.com/32209255/233007413-3fc9f3ce-aa21-4e96-9083-48bb342e0679.png">

This week we are proud to announce the release of Beta 2.0! Big shoutout to our sponsor and contributors that have made all of this possible, you are all awesome! Over the next couple weeks, we will be now focusing on making your agents more powerful! Web search and contextual actions are coming soon!

## ✨  Highlights
* Ability to add your own API key
* Agents can now run on edge and directly in the browser using WASM
* Ability to save and share your agents with others
* Ability to export agent results to pdf, image, or text
* GPT-4 support if you supply your own api key
* Improvement to general UI/UX
* ...and much more!

## :eyes: What's Changed
* :wrench: Update documentation on environment variables by @awtkns in https://github.com/reworkd/AgentGPT/pull/3
* Ability to add your own API key by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/13
* Update FUNDING.yml by @awtkns in https://github.com/reworkd/AgentGPT/pull/14
* Update FUNDING.yml by @awtkns in https://github.com/reworkd/AgentGPT/pull/15
* ⭐ Describe how key is stored by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/31
* [issue-17] Display Copy Icon on Message Hover by @6momo1 in https://github.com/reworkd/AgentGPT/pull/30
* Add Docker support for easier setup and deployment by @Frajder in https://github.com/reworkd/AgentGPT/pull/28
* :sparkles: Improve docker configuration by @awtkns in https://github.com/reworkd/AgentGPT/pull/34
* [Issue-18] Add button to copy text to clipboard by @mrayonnaise in https://github.com/reworkd/AgentGPT/pull/27
* ⚡ Run Agents on vercel/edge by @awtkns in https://github.com/reworkd/AgentGPT/pull/45
* :zap: run all agents on edge by @awtkns in https://github.com/reworkd/AgentGPT/pull/47
* add cd in instructions by @arnav-deep in https://github.com/reworkd/AgentGPT/pull/48
* :hammer: Improve local development setup to be SQLite compatible by @awtkns in https://github.com/reworkd/AgentGPT/pull/49
* Reduce max loops to handle increased load by @reworkd-admin in https://github.com/reworkd/AgentGPT/pull/63
* 👩‍💻 Add markdown formatting by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/70
* Bugfix for #68  by @chozzz in https://github.com/reworkd/AgentGPT/pull/72
* Use regex to filter bad tasks by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/75
* Update README.md by @matigumma in https://github.com/reworkd/AgentGPT/pull/87
* chore: specify node version in readme and package.json by @Jshen123 in https://github.com/reworkd/AgentGPT/pull/89
* Improve Loading Annimations by @awtkns in https://github.com/reworkd/AgentGPT/pull/93
* Added alternative for generating NEXTAUTH_SECRET by @steven-tey in https://github.com/reworkd/AgentGPT/pull/55
* feat(#33): implement model input by @Jshen123 in https://github.com/reworkd/AgentGPT/pull/88
* fix: revert version requirement for node by @Jshen123 in https://github.com/reworkd/AgentGPT/pull/97
* feat: add interfaces, types and constants file by @Jshen123 in https://github.com/reworkd/AgentGPT/pull/96
* feat: Add donation message by @awtkns in https://github.com/reworkd/AgentGPT/pull/103
* :zap: Run agents client side in browser by @awtkns in https://github.com/reworkd/AgentGPT/pull/107
* 📃 Window for current tasks by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/110
* added tailwind css class which fix the bug #98 by @rahul-ghimire-au6 in https://github.com/reworkd/AgentGPT/pull/124
* update readme prisma setup script by @ka-fuachie in https://github.com/reworkd/AgentGPT/pull/118
* fix: remove rogue zero by @ka-fuachie in https://github.com/reworkd/AgentGPT/pull/138
* Add GitHub Codespaces support by @mor10 in https://github.com/reworkd/AgentGPT/pull/135
* Enhance development container configuration with additional features and tools by @exilvm in https://github.com/reworkd/AgentGPT/pull/141
* :lock: Authentication and User Accounts by @awtkns in https://github.com/reworkd/AgentGPT/pull/76
* Create README.md by @Srijan-Subedi in https://github.com/reworkd/AgentGPT/pull/143
* Fix `./setup.sh --docker` build failure by @exilvm in https://github.com/reworkd/AgentGPT/pull/155
* 💅 fix layout overflow by @ka-fuachie in https://github.com/reworkd/AgentGPT/pull/137
* Set correct file permission for setup.sh to be executable by @PeterDaveHello in https://github.com/reworkd/AgentGPT/pull/161
* feat: add combobox for model names by @Jshen123 in https://github.com/reworkd/AgentGPT/pull/106
* feat: refactor combobox to make it more maintainable by @Jshen123 in https://github.com/reworkd/AgentGPT/pull/166
* Update README.md by @eltociear in https://github.com/reworkd/AgentGPT/pull/160
* feat: auto focus on first input on app load by @JNaeemGitonga in https://github.com/reworkd/AgentGPT/pull/139
* feat: improve explanation for GPT4 model requirement by @Jshen123 in https://github.com/reworkd/AgentGPT/pull/172
* bug: improve task styling by @awtkns in https://github.com/reworkd/AgentGPT/pull/180
* feat: add chinese translation for ReadMe by @Jshen123 in https://github.com/reworkd/AgentGPT/pull/188
* feat(#88): add temperature input by @Jshen123 in https://github.com/reworkd/AgentGPT/pull/112
* feat: add max loops setting by @Jshen123 in https://github.com/reworkd/AgentGPT/pull/193
* Update README.md by @awtkns in https://github.com/reworkd/AgentGPT/pull/195
* feat: add loop configuration by @Jshen123 in https://github.com/reworkd/AgentGPT/pull/197
* Feat/add loop configration by @Jshen123 in https://github.com/reworkd/AgentGPT/pull/198
* Pro Model by @awtkns in https://github.com/reworkd/AgentGPT/pull/200
* :bug: Enums not supported in SQLite by @awtkns in https://github.com/reworkd/AgentGPT/pull/206
* Add Dependabot for NPM package updates by @exilvm in https://github.com/reworkd/AgentGPT/pull/210
* build(deps-dev): bump @types/micro from 7.3.7 to 10.0.0 by @dependabot in https://github.com/reworkd/AgentGPT/pull/217
* build(deps-dev): bump postcss from 8.4.21 to 8.4.22 by @dependabot in https://github.com/reworkd/AgentGPT/pull/218
* build(deps): bump langchain from 0.0.56 to 0.0.57 by @dependabot in https://github.com/reworkd/AgentGPT/pull/216
* build(deps-dev): bump @testing-library/react from 13.4.0 to 14.0.0 by @dependabot in https://github.com/reworkd/AgentGPT/pull/215
* build(deps-dev): bump typescript from 4.9.5 to 5.0.4 by @dependabot in https://github.com/reworkd/AgentGPT/pull/214
* Key validation by @matthope1 in https://github.com/reworkd/AgentGPT/pull/219
* Add EditorConfig support by @exilvm in https://github.com/reworkd/AgentGPT/pull/211
* feat/add vscode launch.json by @psgit in https://github.com/reworkd/AgentGPT/pull/142
* fix(#213): update temperature tooltip by @Jshen123 in https://github.com/reworkd/AgentGPT/pull/222
* feat: [#231] Press Enter key to deploy Agent by @giri-madhan in https://github.com/reworkd/AgentGPT/pull/235
* Hungarian translation of the main README.md by @Cs4K1Sr4C in https://github.com/reworkd/AgentGPT/pull/237
* fix: remove english in chinese readme by @Jshen123 in https://github.com/reworkd/AgentGPT/pull/220
* :sparkles: Allow users to save and persist thier agents (WIP) by @awtkns in https://github.com/reworkd/AgentGPT/pull/236
* Saves by @awtkns in https://github.com/reworkd/AgentGPT/pull/247
* bug: fix agent share ui by @awtkns in https://github.com/reworkd/AgentGPT/pull/248

## ❤️ New Contributors
* @awtkns made their first contribution in https://github.com/reworkd/AgentGPT/pull/3
* @asim-shrestha made their first contribution in https://github.com/reworkd/AgentGPT/pull/13
* @6momo1 made their first contribution in https://github.com/reworkd/AgentGPT/pull/30
* @Frajder made their first contribution in https://github.com/reworkd/AgentGPT/pull/28
* @mrayonnaise made their first contribution in https://github.com/reworkd/AgentGPT/pull/27
* @arnav-deep made their first contribution in https://github.com/reworkd/AgentGPT/pull/48
* @reworkd-admin made their first contribution in https://github.com/reworkd/AgentGPT/pull/63
* @chozzz made their first contribution in https://github.com/reworkd/AgentGPT/pull/72
* @matigumma made their first contribution in https://github.com/reworkd/AgentGPT/pull/87
* @steven-tey made their first contribution in https://github.com/reworkd/AgentGPT/pull/55
* @rahul-ghimire-au6 made their first contribution in https://github.com/reworkd/AgentGPT/pull/124
* @ka-fuachie made their first contribution in https://github.com/reworkd/AgentGPT/pull/118
* @mor10 made their first contribution in https://github.com/reworkd/AgentGPT/pull/135
* @exilvm made their first contribution in https://github.com/reworkd/AgentGPT/pull/141
* @Srijan-Subedi made their first contribution in https://github.com/reworkd/AgentGPT/pull/143
* @PeterDaveHello made their first contribution in https://github.com/reworkd/AgentGPT/pull/161
* @eltociear made their first contribution in https://github.com/reworkd/AgentGPT/pull/160
* @JNaeemGitonga made their first contribution in https://github.com/reworkd/AgentGPT/pull/139
* @dependabot made their first contribution in https://github.com/reworkd/AgentGPT/pull/217
* @matthope1 made their first contribution in https://github.com/reworkd/AgentGPT/pull/219
* @psgit made their first contribution in https://github.com/reworkd/AgentGPT/pull/142
* @giri-madhan made their first contribution in https://github.com/reworkd/AgentGPT/pull/235
* @Cs4K1Sr4C made their first contribution in https://github.com/reworkd/AgentGPT/pull/237

**Full Changelog**: https://github.com/reworkd/AgentGPT/compare/v.0.1.0-beta...v.0.2.0-beta

<hr /><em>This discussion was created from the release <a href='https://github.com/reworkd/AgentGPT/releases/tag/v.0.2.0-beta'>v0.2.0 Beta 2.0</a>.</em>

**Comment by ohguinascimento** [comment]
Hi i'm dev and work in my company in free time i invest in inovations, chat gpt is very nice to automatize.

Oi meu nome é Guilherme e eu trabalho pra mim mesmo o chat gpt é muito bom pra automatizar as conversas.

Eu criei o projeto jarvis e vou automatizar usando chat gpt. O projeto jarvis é usado para fazer varredura em redes.

Analise de redes complexas usando automações é muito necessário pra quem é devops.

**Comment by uzuhitris** [comment]
Chat GPT really is a convenient app
